pkg/azure/keyvault/store_test.go

```diff
@@ -124,3 +125,25 @@ func TestParseKeyValueAsSecretID(t *testing.T) {
 		})
 	}
 }
+
+func TestCleanSecretName(t *testing.T) {
+	testcases := map[string]string{
```

Comment:
Can you add a valid secret FQDN as a test case?

Reply:
FQDN values are checked for in Resolve before we call this function, and then are handled separately. So short of an end to end test against resolve (with a live env or mocks), that isn't straightforward to check with the test framework we have in place at the moment.